solr/core/src/java/org/apache/solr/search/facet/FacetParser.java

```diff
@@ -134,9 +138,9 @@ public Object parseFacetOrStat(String key, String type, Object args) throws Synt
     switch (type) {
       case "field":
       case "terms":
-        return new FacetRequest.FacetFieldParser(this, key).parse(args);
+        return new FacetFieldParser(this, key).parse(args);
```

Comment:
Nice; seeing this simple difference makes it clear to me we're now organizing it correctly.